crates/rethnet_eth/src/block/reorg.rs

```diff
@@ -69,3 +70,23 @@ pub fn largest_possible_reorg(chain_id: &U256) -> U256 {
     };
     U256::from(threshold)
 }
+
+/// The interval between blocks for a specific chain.
+pub fn block_time(chain_id: &U256) -> Duration {
+    let chain_id: u64 = chain_id.try_into().expect("invalid chain id");
+    match chain_id {
+        // Ethereum mainnet, Ropsten, Rinkeby, Goerli and Kovan testnets
+        // 32 blocks is one epoch on Ethereum mainnet
+        1 | 3 | 4 | 5 | 42 => Duration::from_secs(12),
+        // Gnosis/xDai
+        // https://gnosisscan.io/chart/blocktime
+        100 => Duration::from_secs(5),
+        _ => {
+            log::warn!(
+                "Unknown chain id {chain_id}, using default block time of {} seconds",
+                rethnet_defaults::DEFAULT_SAFE_BLOCK_TIME.as_secs(),
```

Comment:
In my opinion, the hardcoded Rethnet defaults should only be used in the Rethnet binaries and not be forced upon users.

The only constants we should be using in rethnet_eth are Eth-specific constants. This also avoids circular dependencies, as rethnet_defaults does or would likely depend on rethnet_eth for types.

We already made one temporary addition in a previous PR of yours that will be replaced when Rethnet supports proper config of chains, but we shouldn't keep breaking this pattern

Reply:
I agree in principle, but I see no other way to do it in this PR. Can we fix this at the same time as the other default (safety depth)?

Reply:
Sure